pkg/virtctl/config_ssh/config_ssh.go

```diff
@@ -0,0 +1,221 @@
+)
+
+const (
+	CONFIG_SSH_COMMAND = "config-ssh"
+	KubeVirtEOLComment = "Generated by KubeVirt"
+)
+
+var (
+	dryRun        bool
+	removeEntries bool
+	sshConfigFile string
+)
+
+func NewConfigSSHCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "config-ssh",
+		Short: "Write VM(I)s into ~/.ssh/config.",
+		Long: `Write VM(I)s into .ssh/config to get convenient aliases in ~/.ssh/config for quickly accessingVMs with ssh.
+On most system this will give auto-completion for ssh invocations.
+`,
+		Example: examples(),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := VirtCommand{
+				command:      CONFIG_SSH_COMMAND,
+				clientConfig: clientConfig,
+			}
+			return c.Run(args)
+		},
+	}
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	cmd.Flags().BoolVar(&dryRun, "dry-run", false, "Show ssh config changes instead of writing them to the file.")
+	cmd.Flags().BoolVar(&removeEntries, "remove", false, "Remove all host entries from the ssh config file added by KubeVirt.")
+	cmd.Flags().StringVar(&sshConfigFile, "ssh-config-file", "", "Specifies an alternative per-user SSH configuration file. By default, this is ~/.ssh/config.")
+	return cmd
+}
+
+type VirtCommand struct {
+	clientConfig clientcmd.ClientConfig
+	command      string
+}
+
+func (vc *VirtCommand) Run(args []string) error {
+
+	programName := templates.GetProgramName(filepath.Base(os.Args[0]))
+
+	if sshConfigFile == "" {
+		home, err := os.UserHomeDir()
+		if err != nil {
+			return fmt.Errorf("could not determine home directory of the user: %v", err)
+		}
+		sshConfigFile = filepath.Join(home, ".ssh", "config")
+	}
+	cfg, fileMode, err := loadSSHConfig(sshConfigFile)
+	if err != nil {
+		return err
+	}
+
+	cfg.Hosts = removeHostEntries(cfg.Hosts)
+
+	if !removeEntries {
+		namespace, _, err := vc.clientConfig.Namespace()
+		if err != nil {
+			return err
+		}
+
+		virtClient, err := kubecli.GetKubevirtClientFromClientConfig(vc.clientConfig)
+		if err != nil {
+			return fmt.Errorf("cannot obtain KubeVirt client: %v", err)
+		}
+		rawConfig, err := vc.clientConfig.RawConfig()
+		if err != nil {
+			return fmt.Errorf("failed to determine current context: %v", err)
+		}
+		currentContext := rawConfig.CurrentContext
+
+		for _, gvr := range []schema.GroupVersionResource{k6sv1.VirtualMachineInstanceGroupVersionResource, k6sv1.VirtualMachineGroupVersionResource} {
+			objects, err := virtClient.DynamicClient().Resource(gvr).Namespace(namespace).List(context.Background(), v1.ListOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to fetch %s: %v", gvr.Resource, err)
+			}
+
+			hosts, err := generateHostEntries(programName, currentContext, objects.Items)
+			if err != nil {
+				return err
+			}
+			cfg.Hosts = append(cfg.Hosts, hosts...)
+		}
+	}
+
+	if dryRun {
+		fmt.Println(cfg.String())
+		return nil
+	} else {
+		return ioutil.WriteFile(sshConfigFile, []byte(cfg.String()), fileMode)
```

Comment:
Are we sure that that ssh_config lib is not missing any features so a round trip is safe with any config?
If not we should probably save a backup somewhere.

Reply:
I could not find anything in the parser which would modify or drop entries. `mech` entries are not supported, but there the parser just fails with an error.

Another possibility would be to add a single `Include` directive to the file an manage all `kubevirt` entries in a separate file which gets included.

Reply:
I like the `Include` idea. That would declutter the config file for people that might interact more with it.